rust-runtime/aws-smithy-http-server-python/src/tls.rs

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! TLS related types for Python.
+//!
+//! [PyTlsConfig] implementation is mostly borrowed from:
+//! <https://github.com/seanmonstar/warp/blob/4e9c4fd6ce238197fd1088061bbc07fa2852cb0f/src/tls.rs>
+
+use std::fs::File;
+use std::io::{self, BufReader, Read};
+use std::time::Duration;
+
+use pyo3::{pyclass, pymethods};
+use thiserror::Error;
+use tokio_rustls::rustls::{Certificate, Error as RustTlsError, PrivateKey, ServerConfig};
+
+pub mod listener;
+
+/// PyTlsConfig represents TLS configuration created from Python.
+#[pyclass(
+    name = "TlsConfig",
+    text_signature = "(*, key_path, cert_path, reload)"
+)]
+#[derive(Clone)]
+pub struct PyTlsConfig {
+    /// Absolute path of the RSA or PKCS private key.
+    key_path: String,
```

Comment:
Question since I'm not too familiar with Rust choices: Why `String` rather than `std::path::Path`?

Reply:
Good point! No reason, I've changed it to use `PathBuf` which is owned version of `Path`